### DISCUSSION TITLE
划词翻译

### DISCUSSION THREAD

**Discussion by Kenshin** [Kenshin/simpread#1592] [插件说明]
描述
---

>  划词翻译，包括： 单词、整句，并支持生词本功能。

地址
---

[https://simpread.ksria.cn/plugins/details/ohnTKVHz4a](https://simpread.ksria.cn/plugins/details/ohnTKVHz4a)

视频
---

https://user-images.githubusercontent.com/81074/187348845-d36e46f0-c9ec-4f30-b6b5-d2078fbcafbd.mp4

内置多种翻译引擎
---

内置与 [全文翻译](https://github.com/Kenshin/simpread/discussions/1593) 一样的多种翻译引擎，所以，关于这些翻译 API Token 的获取方案来自全文翻译的说明。

- [微软翻译](https://github.com/Kenshin/simpread/discussions/1593#discussioncomment-6149415) **默认翻译引擎，无 Token 无需登录**

- [谷歌翻译](https://github.com/Kenshin/simpread/discussions/1593#discussioncomment-6158615) **无 Token 无需登录**

- [百度翻译](https://github.com/Kenshin/simpread/discussions/1593#discussioncomment-1147772)

- [腾讯翻译君](https://github.com/Kenshin/simpread/discussions/1593#discussioncomment-1147773)

- [彩云小译](https://github.com/Kenshin/simpread/discussions/1593#discussioncomment-1152241)

- [小牛翻译](https://github.com/Kenshin/simpread/discussions/1593#discussioncomment-1156896)

- [DeepL](https://github.com/Kenshin/simpread/discussions/1593#discussioncomment-3492095)

- [OpenAI](https://github.com/Kenshin/simpread/discussions/1593#discussioncomment-5205592)

其中百度翻译内置了 Token，是简悦官方内置测试用，**随时有可能因使用额度导致被停止的问题**，请更换为自己的 Token，如何申请 [请看这里](https://github.com/Kenshin/simpread/discussions/1593#discussioncomment-1147772)。

![image](https://user-images.githubusercontent.com/81074/187071560-98c47e6e-7bc3-44cc-a619-55913620a463.png)

生词本
---

生词本是 1.0.0 增加的功能，也是最重要的功能，通过生词本，可以快速标记当前的生词，并且自动显示翻译结果，详细说明 [请看这里](https://github.com/Kenshin/simpread/discussions/1592#discussioncomment-3499298)。

![](https://user-images.githubusercontent.com/81074/187338450-d4d4285a-3c3c-4e6b-bf00-dd8e42b612b5.png)

选项
---

随便进入一个页面 e.g. https://sspai.com/post/71576 并进入 **阅读模式 → 右下角 → 动作 → 插件触发器**

按照下图所示即可查看此插件的选项页。（注意：如果你的界面没有插件触发器，请根据此 [解决方案](https://github.com/Kenshin/simpread/discussions/2342) 修复。

<img width="1262" alt="image" src="https://github.com/Kenshin/simpread/assets/81074/a3b9ca62-0698-4fc9-91aa-725258015b28">

功能
---

- 支持多种翻译引擎

- 划词翻译

    ![2023-06-11_15-42-51 (1)](https://github.com/Kenshin/simpread/assets/81074/1469c5e7-027e-411c-9482-f6254927d478)

- 划词后以何种方式翻译，包括：鼠标移动上 · 鼠标点击 · 自动翻译。

   ![image](https://user-images.githubusercontent.com/81074/187345885-31143342-8aef-4f88-96ad-e0828c5ad9dc.png)

- [生词本](https://github.com/Kenshin/simpread/discussions/1592#discussioncomment-3499298)

- OpenAI 支持 [自定义 API（来自全文翻译的文档说明）](https://github.com/Kenshin/simpread/discussions/1593#discussioncomment-6389285) 与 [自定义模型（来自阅读助手的文档说明）](https://github.com/Kenshin/simpread/discussions/5158#discussioncomment-10354289)

安装与升级
---

- 没有【使用同步助手 · 自动同步】的话，可通过[此链接](https://simpread.ksria.cn/plugins/details/ohnTKVHz4a) 安装。

- 已经【使用同步助手 · 自动同步】的话，请使用 [插件管理器](https://www.yuque.com/kenshin/simpread/fcw60u#ymS1C) 安装。

更新日志
---

### 2.2.2

- `Add` 2025年4月14日，GPT 更新了最新的 `4.1` 系列模型，包括： `gpt-4.1-mini` `gpt-4.1-nano` `gpt-4.1`

- `Remove` GPT 旧模型，包括：`gpt-3.5-turbo` `gpt-4o-mini` `gpt-4o` `gpt-4` `gpt-4-turbo` `gpt-4-32k` （当使用这些模型时，会自动转换为 `4.1` 系列模型）

### 2.2.1

- `Add` [对任意 API 结构的支持](https://github.com/Kenshin/simpread/discussions/1593#discussioncomment-6389285)，即：API 可以跟 OpenAI 不一致，但返回结构必须要与 OpenAI 一致，支持三种方式：

    1. 主域名方式 e.g. `api.siliconflow.cn` 
    
    2. 全 URL 方式 e.g. `https://openrouter.ai/api` 
    
    3. 非 OpenAI 结构，需要使用 `#` 强制使用 e.g. `https://openrouter.ai/api/v1#`    

### 2.2.0

- `Remove` ChatGPT（因 API 改变导致无法使用此方式）

- `Add`  [自定义模型](https://github.com/Kenshin/simpread/discussions/5158#discussioncomment-5175855)

- `Add` 版本检查（选项卡左下角版本号，点击可验证是否有新版）

### 2.1.0

- `Add` OpenAI stream 方案

- `Add` 在浮动工具条注入一个新的 Icon

- `Optimize` 选项卡 UI 升级到 2.0

- `Fix` 生词本白名单为空的判断错误

### 2.0.2

- `Optimize` `OpenAI` 统一更换为 `gpt-4o-mini`

- `Optimize` `OpenAI` 更换 Prompt 并增加 `网络释义`

- `Remove` ChatGPT Web 版 API

### 2.0.1

- `Optimize` 自定义 API 时仅输入主域名，与 [阅读助手](https://github.com/Kenshin/simpread/discussions/5158) 保持一致。

### 2.0.0

- `Add` 新的 [选项卡](https://github.com/Kenshin/simpread/assets/81074/a3b9ca62-0698-4fc9-91aa-725258015b28)

- `Add` OpenAI 官方源与自定义源

- `Add` [ChatGPT 源](https://github.com/Kenshin/simpread/discussions/1593#discussioncomment-6119141)

- `Add` [微软翻译](https://github.com/Kenshin/simpread/discussions/1593#discussioncomment-6149415)

- `Add` [谷歌翻译](https://github.com/Kenshin/simpread/discussions/1593#discussioncomment-6158615)

- `Add` 当设置完翻译引擎后的测试功能

- `Add` `Loading` 状态可取消

- `Add` 划词解释（仅限 ChatGPT）    
    > 当划词（只要）包含中文时，不翻译而直接解释。

- `Add` 标注时可以划词

- `Add` `gpt-3.5-turbo-0613` 模型

- `Add` 划词逻辑以及对 `标题`和 `描述` 的划词监控

### 1.0.5

- `Fix` 腾讯翻译在划取短语时的调用错误，关联 https://github.com/Kenshin/simpread/issues/5607

### 1.0.4

- `Add`  接入 [OpenAI](https://github.com/Kenshin/simpread/discussions/1593#discussioncomment-5205592)

### 1.0.3

- `Fix` 生词本的添加/删除逻辑。

- `Optimize` 翻译引擎翻译时的调用逻辑。

### 1.0.2

- `Fix` 百度翻译的若干问题。

### 1.0.1 

- `Fix` 划词时的提示信息。

### 1.0.0

- `Add` 多种翻译引擎

- `Add`选项页

    - 划词后以何种方式翻译，包括：鼠标移动上 · 鼠标点击 · 自动翻译。

- `Add` [生词本](https://github.com/Kenshin/simpread/discussions/1592#discussioncomment-3499298)

关联
---

简悦适配了很多关于英语学习的技巧，详细 [请看这里](https://github.com/Kenshin/simpread/discussions?discussions_q=label%3Alanguage)。

- [划词翻译](https://github.com/Kenshin/simpread/discussions/1592)

- [英文阅读版面辅助增强](https://github.com/Kenshin/simpread/discussions/2841)

- [高级版统计（用于英文阅读）](https://github.com/Kenshin/simpread/discussions/1590)

- [使用彩云小译实现双语翻译并同时保存在本地](https://github.com/Kenshin/simpread/discussions/3182)

- [适配扇贝阅读](https://github.com/Kenshin/simpread/discussions/2705)

**Comment by swithun-liu** [comment]
和标注的那个bar有冲突，希望能和标注bar集成在一起

（其他的浏览器翻译插件在阅读模式下无法使用）

**Comment by fedono** [comment]
可以增加一个设置？选中单词就自动显示翻译，取消选中后，就自动隐藏掉翻译？就不用鼠标移动到「点我翻译」再显示，就可以少一步操作了 @Kenshin

**Reply by Kenshin** [reply]
okay 我考虑下。 可能有点误操作的问题，但需求我理解。

**Reply by Kenshin** [reply]
嗨，划词翻译 1.0.0 版发布了，增加了这个功能，关于此版本的更新说明

- `Add` 多种翻译引擎

- `Add`选项页

    - 划词后以何种方式翻译，包括：鼠标移动上 · 鼠标点击 · 自动翻译。

以及本次更新最重要的一个功能：[**生词本**](https://github.com/Kenshin/simpread/discussions/1592#discussioncomment-3499298)，通过此功能，可以快速标记当前的生词，并且自动显示翻译结果。

**Comment by Kenshin** [comment]
# 生词本

生词本是 1.0.0 增加的功能，也是最重要的功能，通过生词本，可以快速标记当前的生词，并且自动显示翻译结果。

## 视频

https://user-images.githubusercontent.com/81074/187336464-ea721844-1fc3-4222-a827-5b87d0cac3ee.mp4

## 功能

- [进入阅读模式后自动加载生词本与白名单](https://github.com/Kenshin/simpread/discussions/1592#discussioncomment-3505167)

- [生词本后的加载方式](https://github.com/Kenshin/simpread/discussions/1592#discussioncomment-3499395)

- [自动翻译](https://github.com/Kenshin/simpread/discussions/1592#discussioncomment-3499410)

- [生词本的使用](https://github.com/Kenshin/simpread/discussions/1592#discussioncomment-3505230)

- [生词本的制作](https://github.com/Kenshin/simpread/discussions/1592#discussioncomment-3505300)

- [添加/删除生词本](https://github.com/Kenshin/simpread/discussions/1592#discussioncomment-3505374)

**Reply by Kenshin** [reply]
# 生词本后的加载方式

当使用生词本的加载方式，包括：自动加载 与 触发器加载。如果你的生词本内容较少，比如少于 100 个，可使用自动加载，反之，建议使用触发器方案。

## 选项

![image](https://user-images.githubusercontent.com/81074/187337792-41a8f7aa-0210-44e8-ab87-6918e60989c4.png)

## 触发器

安装划词翻译后，会在右下角出现一个 Icon，点击后可自动加载生词本，再点击一次，可删除生词本。

![image](https://user-images.githubusercontent.com/81074/187337503-bf385d24-7b5d-48fe-9c94-57599057a1bf.png)

## 注意

此功能受限于 [白名单](https://github.com/Kenshin/simpread/discussions/1592#discussioncomment-3505167) 功能。

**Reply by Kenshin** [reply]
# 自动翻译

如果你的生词本只有单词，但没有翻译的话，可启用此功能。

## 选项

![image](https://user-images.githubusercontent.com/81074/187337873-3ff09ab1-02e0-4a33-bfe6-001b8f3d9844.png)

## 用法

关闭选项后，仅保留生词本的单词，方便学习。

![image](https://user-images.githubusercontent.com/81074/187338281-29adb7ba-f6ab-4b83-9f36-b24c7c0dbfe9.png)

开启选项后

![image](https://user-images.githubusercontent.com/81074/187338450-d4d4285a-3c3c-4e6b-bf00-dd8e42b612b5.png)

## 注意

开启后，如果生词本对应的内容没有翻译，会自动调用翻译引擎对应的 API 进行翻译，如果生词没有翻译的单词较多，则翻译时间会变成长。

**Reply by Kenshin** [reply]
# 添加或删除生词本

在阅读模式下划词翻译后，可将其添加 /  删除当前划取的单词。

## 视频操作

https://user-images.githubusercontent.com/81074/187345733-bac21ca8-369f-4c56-a347-af3abb674074.mp4

**Comment by Kenshin** [comment]
# 进入阅读模式后自动加载生词本与白名单

这是生词本的功能，为了方便在不相干的页面不使用生词本，所以增加了白名单机制，包括：**自动启用生词本** 和 **白名单** 机制。

## 选项

![image](https://user-images.githubusercontent.com/81074/187338990-246043ae-b4f5-4007-a06a-212c42ef134c.png)

## 进入阅读模式后是否自动加载生词本

开启后，无论进入什么页面都会自动加载生词本，不建议开启。

## 白名单

当前关闭 ** 进入阅读模式后是否自动加载生词本** 后，就可以使用白名单了，支持

- 全 URL e.g. `https://web.shanbay.com/reading/web-news/articles/bvxsus`
- 主域名 e.g. sspai.com （会自动忽略 `www` ）

一个例子

```
sspai.com
https://web.shanbay.com/reading/web-news/articles/bvxsus
```

也就是当访问 `sspai.com` 或者 `https://web.shanbay.com/reading/web-news/articles/bvxsus` 后才启用全文翻译。

**Comment by Kenshin** [comment]
# 生词本的用法

只有生词本中存在单词，才会将当前页面的对应的单词标记出来。

## 用法

每行一个，格式为 `<单词>,<翻译>` 或 `<单词>` ，也就是说 `<翻译>` 可为空，当为空时，`,` 也需要省略。 e.g.

- `wordy,adj: 罗嗦，啰嗦，冗长` 
- `Delete`

## 例子

将下面的内容加入到生词本

```
Delete
wordy,adj: 罗嗦，啰嗦，冗长,'wә:di
carve,vt. 雕刻；切开；开创 vi. 切开；做雕刻工作
edit
complicated
```

如下图

![image](https://user-images.githubusercontent.com/81074/187340386-a5c996b0-da3d-47b2-a47d-204f132c6e3f.png)

进入 https://web.shanbay.com/reading/web-news/articles/bvxsus 然后加载生词本后会得到下图的效果

![image](https://user-images.githubusercontent.com/81074/187340286-9c601a68-5794-4546-8ee7-6ac469ee2362.png)

## 注意

当 `<翻译>` 为空时，并且启用了 [自动翻译](https://github.com/Kenshin/simpread/discussions/1592#discussioncomment-3499410)
 则会调用之前设定的翻译 API 进行翻译。

**Comment by Kenshin** [comment]
# 生词本的制作

生词本的制作比较简单，只要符合：每行一个，格式为 `<单词>,<翻译>` 或 `<单词>` 即可。

但网络上有很多现成的单词表，可以方便的转换为上述格式。

## 来源

- https://github.com/mahavivo/english-wordlists

- https://github.com/ourongxing/maimemo-export/tree/main/thesaurus%2Fcsv

## 如何制作

一般来说，如果是 `.txt` 格式的话，大多会符合简悦的格式规则，现在说下 `.csv` 格式的生词本。

## 例子

打开 [这个链接](https://github.com/ourongxing/maimemo-export/blob/main/thesaurus/csv/17%E5%A4%A9%E6%90%9E%E5%AE%9AGRE%E5%8D%95%E8%AF%8D.csv) 然后看下面的视频操作

https://user-images.githubusercontent.com/81074/187343028-886fa39f-00cf-4f87-9102-a8d5b51c24ba.mp4

## 官方示例

https://github.com/Kenshin/simpread/discussions/1592#discussioncomment-3505331

## 注意

如果生词本超过 2K 行的话，会比较消耗性能，建议以 2K 以内。

**Reply by Kenshin** [reply]
# 官方示例

简悦官方提供了几个转换好的生词本，都是来自与上面提到的两个链接，由简悦官方转换。

- [测试用例.txt](https://github.com/Kenshin/simpread/files/9449386/default.txt) 专门用于测试 https://web.shanbay.com/reading/web-news/articles/bvxsus 而定制的生词本。

- [17天搞定GRE单词.txt](https://github.com/Kenshin/simpread/files/9449379/17.GRE.txt)

- [2011GRE核心词汇考法精析.txt](https://github.com/Kenshin/simpread/files/9449380/2011GRE.txt)

- [六级词汇表.txt](https://github.com/Kenshin/simpread/files/9449381/default.txt)

- [GRE核心词汇考法精析.txt](https://github.com/Kenshin/simpread/files/9452366/GRE.txt)

**Reply by Kenshin** [reply]
# CEFR

再分享几个 **CEFR（ 全名 Common European Framework of Reference for Language）** 制作好的词汇表，来自简悦用户的分享。

- [CERF B1.txt](https://github.com/Kenshin/simpread/files/9486514/CERF.B1.txt)

- [CERF C1.txt](https://github.com/Kenshin/simpread/files/9486515/CERF.C1.txt)

- [CERF C2.txt](https://github.com/Kenshin/simpread/files/9486516/CERF.C2.txt)

**Comment by Kenshin** [comment]
# 如何在标注模式下划词

## 描述

在之前的版本中 **当开启标注后无法进行划词操作**，从 `2.0.0` 版开始支持 **在标注模式下划词**

## 如何设置

不需要做任何事情，仅需要确保你的划词翻译版本为 **2.0.0** 即可。

## 更好的方案

2.1.0 版增加了在 **标注浮动工具条** 增加一个 Icon 来替代原来的方式

<img width="483" alt="image" src="https://github.com/user-attachments/assets/f91f3fe4-dd5b-435a-bb7c-87f3ff181b22">

开启选项后效果如下

<img width="382" alt="dict@origin" src="https://github.com/user-attachments/assets/db7fd670-e732-4c21-a2dc-432dc878a04c">

留意下图

![dict-inject-tips](https://github.com/user-attachments/assets/7cf296fe-24fc-4c69-b474-80765d640569)

## 使用建议

如果你经常使用划词翻译的话，建议 **取消连续标注** 功能。（位置在：选项页 → 高级设定 → 标注）

<img width="1184" alt="image" src="https://github.com/Kenshin/simpread/assets/81074/989346cc-fbb8-4a08-ba9d-74cf33108994">

## 演示

https://github.com/Kenshin/simpread/assets/81074/b01ed8b7-b2e4-4b7f-9cb3-371ce8cb61a4
